examples/README.md

```diff
@@ -9,31 +9,117 @@ This collection of sample templates will help you get started with MongoDB Atlas
+To create the execution role, do one of the following steps:
+- Use the [execution-role.yaml](execution-role.yaml) template
+- Create the execution-role as part of the [next step](#activate-the-mongodb-atlas-cloudformation-public-extensions)
+- Use the [AWS Console](https://us-east-1.console.aws.amazon.com/iam/home?region=us-east-1#/roles) or another method of your choice.
 
+### Activate the MongoDB Atlas CloudFormation public extensions
+You have the following options for activating the extensions, for example `MongoDB::Atlas::Cluster`:
 
-### CloudFormation Profile
-A profile should be created in the AWS Secrets Manager, containing the MongoDB Atlas Programmatic API Key.
+- Use the [CloudFormation service in the AWS Console](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/registry/public-extensions?visibility=PUBLIC&type=RESOURCE&category=THIRD_PARTY):
+   1. Ensure you are in the correct AWS region.
+   2. Select Publisher=`ThirdParty`.
+   3. Extension name prefix = `MongoDB::Atlas`.
+   4. Select the resource type and click `Activate`.
+   5. In the *Execution role ARN* specify the role arn, for example `arn:aws:iam::123456789012:role/cfn-execution-role`, from the previous step.
+- Use the provided [CFN template](activate-mongodb-atlas-resources.template.yaml) to create an IAM execution role and activate all MongoDB Atlas extensions:
+   1. Specify the Region.
+   2. [Create the stack](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks?filteringText=&filteringStatus=active&viewNested=true). If the stack has been run before in the same region, you will need to delete the previous stack and re-create the stack to re-activate the extensions.
 
-Use [this template](profile-secret.yaml) to create a [new CloudFormation stack](https://console.aws.amazon.com/cloudformation/home#/stacks/create) for the default profile that all resources will attempt to use unless a different override is specified.
+### Configure your KMS Key Policy
+If your profile secret is encrypted with a KMS Customer Managed Key (CMK), you must configure the key to allow access.
+To give the `execution role` access to the CMK, you must configure a [KMS Key Policy](https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html).
+The recommended options include:
+- [Allow access via Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/security-encryption.html#security-encryption-policies):
+  ```json
+  {
+    "Statement": [
+        // other statements
+                {
+            "Sid": "Enable IAM User Permissions",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::123456789:root"
+            },
+            "Action": "kms:*",
+            "Resource": "*"
+        },
```

Comment:
For this first option, is it necessary to enable IAM user permissions for the key policy?

Reply:
Yes. The key must have at least one admin. Otherwise, It is not possible to update the key policy.
However, it can be more restrictive. Added a comment

Reply:
thanks you! great follow up additions to make first option clear.